lls/cache.c

```diff
@@ -408,9 +474,15 @@ process_ping_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
 
 	num_granted_pkts = tb_ratelimit_allow_n(num_pkts, rs);
 
-	for (i = 0; i < num_granted_pkts; i++)
-		(*xmit_reply_fn)(iface, pkts[i]);
-	send_pkts(iface->id, tx_queue, num_granted_pkts, pkts);
+	while (num_ping_reply_pkts < num_granted_pkts && i < num_pkts) {
```

Comment:
Style: rewriting this `while` loop as a `for` loop emphasizes that the following loop is a continuation of this loop. After changing it, remember to drop the initialization of `i` done above, and the increment of `i` done in the body of the loop.